```
cmd.exe /C "cd . && T:\1\bin\swiftc.exe -output-file-map Sources\X509\CMakeFiles\X509.dir\Release\output-file-map.json -incremental -j 36 -emit-library -static -o lib\X509.lib -module-name X509 -module-link-name X509 -emit-module -emit-module-path swift\X509.swiftmodule -emit-dependencies  -O -Xcc -FT:/4 -I T:\4\swift -I T:\12\swift -I C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-crypto\Sources\CCryptoBoringSSL\include -I T:\14\swift @CMakeFiles\X509.rsp   && cd ."
C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-certificates\Sources\X509\Certificate.swift:15:8: error: missing required modules: 'CoreFoundation', 'Dispatch'
```

--- a/utils/build-windows-toolchain.bat
+++ b/utils/build-windows-toolchain.bat
@@ -553,9 +553,61 @@ cmake ^
 cmake --build %BuildRoot%\13 || (exit /b)
 cmake --build %BuildRoot%\13 --target install || (exit /b)
 
+:: Build swift-asn1
+ cmake ^
+   -B %BuildRoot%\14 ^
+
+   -D BUILD_SHARED_LIBS=NO ^
+   -D CMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
+   -D CMAKE_C_COMPILER=%BuildRoot%/1/bin/clang-cl.exe ^
+   -D CMAKE_C_FLAGS="/GS- /Oy /Gw /Gy" ^
+   -D CMAKE_CXX_COMPILER=%BuildRoot%/1/bin/clang-cl.exe ^
+   -D CMAKE_CXX_FLAGS="/GS- /Oy /Gw /Gy" ^
+   -D CMAKE_MT=mt ^
+   -D CMAKE_Swift_COMPILER=%BuildRoot%/1/bin/swiftc.exe ^
+   -D CMAKE_EXE_LINKER_FLAGS="/INCREMENTAL:NO" ^
+   -D CMAKE_SHARED_LINKER_FLAGS="/INCREMENTAL:NO" ^
+
+   -D CMAKE_INSTALL_PREFIX=%InstallRoot% ^
+
+   -D dispatch_DIR=%BuildRoot%\3\cmake\modules ^
+   -D Foundation_DIR=%BuildRoot%\4\cmake\modules ^
+
+   -G Ninja ^
+   -S %SourceRoot%\swift-asn1 || (exit /b)
+ cmake --build %BuildRoot%\14 || (exit /b)
+ cmake --build %BuildRoot%\14 --target install || (exit /b)
+
+:: Build swift-certificates
+ cmake ^
+   -B %BuildRoot%\15 ^
+
+   -D BUILD_SHARED_LIBS=NO ^
+   -D CMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
+   -D CMAKE_C_COMPILER=%BuildRoot%/1/bin/clang-cl.exe ^
+   -D CMAKE_C_FLAGS="/GS- /Oy /Gw /Gy" ^
+   -D CMAKE_CXX_COMPILER=%BuildRoot%/1/bin/clang-cl.exe ^
+   -D CMAKE_CXX_FLAGS="/GS- /Oy /Gw /Gy" ^
+   -D CMAKE_MT=mt ^
+   -D CMAKE_Swift_COMPILER=%BuildRoot%/1/bin/swiftc.exe ^
+   -D CMAKE_EXE_LINKER_FLAGS="/INCREMENTAL:NO" ^
+   -D CMAKE_SHARED_LINKER_FLAGS="/INCREMENTAL:NO" ^
+
+   -D CMAKE_INSTALL_PREFIX=%InstallRoot% ^
+
+   -D dispatch_DIR=%BuildRoot%\3\cmake\modules ^
+   -D Foundation_DIR=%BuildRoot%\4\cmake\modules ^
+   -D SwiftCrypto_DIR=%BuildRoot%\12\cmake\modules ^
+   -D SwiftASN1_DIR=%BuildRoot%\14\cmake\modules ^
+
+   -G Ninja ^
+   -S %SourceRoot%\swift-certificates || (exit /b)
+ cmake --build %BuildRoot%\15 || (exit /b)
+ cmake --build %BuildRoot%\15 --target install || (exit /b)
+
 :: Build swift-package-manager
 cmake ^
-  -B %BuildRoot%\14 ^
+  -B %BuildRoot%\16 ^
 
   -D CMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
   -D CMAKE_C_COMPILER=%BuildRoot%/1/bin/clang-cl.exe ^
@@ -579,17 +631,19 @@ cmake ^
   -D SwiftDriver_DIR=%BuildRoot%\11\cmake\modules ^
   -D SwiftCrypto_DIR=%BuildRoot%\12\cmake\modules ^
   -D SwiftCollections_DIR=%BuildRoot%\13\cmake\modules ^
+  -D SwiftASN1_DIR=%BuildRoot%\14\cmake\modules ^
+  -D SwiftCertificates_DIR=%BuildRoot%\15\cmake\modules ^
   -D SQLite3_INCLUDE_DIR=%BuildRoot%\Library\sqlite-3.36.0\usr\include ^
   -D SQLite3_LIBRARY=%BuildRoot%\Library\sqlite-3.36.0\usr\lib\SQLite3.lib ^
 
   -G Ninja ^
   -S %SourceRoot%\swiftpm || (exit /b)
-cmake --build %BuildRoot%\14 || (exit /b)
-cmake --build %BuildRoot%\14 --target install || (exit /b)
+cmake --build %BuildRoot%\16 || (exit /b)
+cmake --build %BuildRoot%\16 --target install || (exit /b)
 
 :: Build IndexStoreDB
 cmake ^
-  -B %BuildRoot%\15 ^
+  -B %BuildRoot%\17 ^
 
   -D BUILD_SHARED_LIBS=NO ^
   -D CMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
@@ -609,11 +663,11 @@ cmake ^
 
   -G Ninja ^
   -S %SourceRoot%\indexstore-db || (exit /b)
-cmake --build %BuildRoot%\15 || (exit /b)
+cmake --build %BuildRoot%\17 || (exit /b)
 
 :: Build swift-syntax
 cmake ^
-  -B %BuildRoot%\16 ^
+  -B %BuildRoot%\18 ^
 
   -D CMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
   -D CMAKE_C_COMPILER=%BuildRoot%/1/bin/clang-cl.exe ^
@@ -627,12 +681,12 @@ cmake ^
 
   -G Ninja ^
   -S %SourceRoot%\swift-syntax || (exit /b)
-cmake --build %BuildRoot%\16 || (exit /b)
-cmake --build %BuildRoot%\16 --target install || (exit /b)
+cmake --build %BuildRoot%\18 || (exit /b)
+cmake --build %BuildRoot%\18 --target install || (exit /b)
 
 :: Build SourceKit-LSP
 cmake ^
-  -B %BuildRoot%\17 ^
+  -B %BuildRoot%\19 ^
 
   -D BUILD_SHARED_LIBS=YES ^
   -D CMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
@@ -654,15 +708,15 @@ cmake ^
   -D LLBuild_DIR=%BuildRoot%\8\cmake\modules ^
   -D ArgumentParser_DIR=%BuildRoot%\9\cmake\modules ^
   -D Yams_DIR=%BuildRoot%\10\cmake\modules ^
-  -D SwiftPM_DIR=%BuildRoot%\14\cmake\modules ^
-  -D IndexStoreDB_DIR=%BuildRoot%\15\cmake\modules ^
+  -D SwiftPM_DIR=%BuildRoot%\16\cmake\modules ^
+  -D IndexStoreDB_DIR=%BuildRoot%\17\cmake\modules ^
   -D SwiftCollections_DIR=%BuildRoot%\13\cmake\modules ^
-  -D SwiftSyntax_DIR=%BuildRoot%\16\cmake\modules ^
+  -D SwiftSyntax_DIR=%BuildRoot%\18\cmake\modules ^
 
   -G Ninja ^
   -S %SourceRoot%\sourcekit-lsp || (exit /b)
-cmake --build %BuildRoot%\17 || (exit /b)
-cmake --build %BuildRoot%\17 --target install || (exit /b)
+cmake --build %BuildRoot%\19 || (exit /b)
+cmake --build %BuildRoot%\19 --target install || (exit /b)
 
 :: Create Configuration Files
 python -c "import plistlib; print(str(plistlib.dumps({ 'DefaultProperties': { 'DEFAULT_USE_RUNTIME': 'MD' } }), encoding='utf-8'))" > %SDKInstallRoot%\SDKSettings.plist

